docs/source/usage_guides/model_size_estimator.md

```diff
@@ -0,0 +1,134 @@
+│float16│   42.47 MB  │206.59 MB │     826.36 MB     │
+│  int8 │   21.24 MB  │103.29 MB │     413.18 MB     │
+│  int4 │   10.62 MB  │ 51.65 MB │     206.59 MB     │
+└───────┴─────────────┴──────────┴───────────────────┘
+```
+
+By default it will return all the supported dtypes (`int4` through `float32`), but if you are interested in specific ones these can be filtered.
+
+### Specific libraries
+
+If the source library cannot be determined automatically (like it could in the case of `bert-base-cased`), a library name can
+be passed in. 
+
+```bash
+accelerate estimate-memory HuggingFaceM4/idefics-80b-instruct --library_name transformers
+```
+
+```
+┌────────────────────────────────────────────────────────────────────┐
+│   Memory Usage for loading `HuggingFaceM4/idefics-80b-instruct`    |
+├───────┬─────────────┬──────────┬───────────────────────────────────┤
+│ dtype │Largest Layer│Total Size│        Training using Adam        │
+├───────┼─────────────┼──────────┼───────────────────────────────────┤
+│float32│   3.02 GB   │297.12 GB │              1.16 TB              │
+│float16│   1.51 GB   │148.56 GB │             594.24 GB             │
+│  int8 │  772.52 MB  │ 74.28 GB │             297.12 GB             │
+│  int4 │  386.26 MB  │ 37.14 GB │             148.56 GB             │
+└───────┴─────────────┴──────────┴───────────────────────────────────┘
+```
+
+```bash
+accelerate estimate-memory timm/resnet50.a1_in1k --library_name timm
+```
+
+```
+┌────────────────────────────────────────────────────┐
+│ Memory Usage for loading `timm/resnet50.a1_in1k`   │
+├───────┬─────────────┬──────────┬───────────────────┤
+│ dtype │Largest Layer│Total Size│Training using Adam│
+├───────┼─────────────┼──────────┼───────────────────┤
+│float32│    9.0 MB   │ 97.7 MB  │     390.78 MB     │
+│float16│    4.5 MB   │ 48.85 MB │     195.39 MB     │
+│  int8 │   2.25 MB   │ 24.42 MB │      97.7 MB      │
+│  int4 │   1.12 MB   │ 12.21 MB │      48.85 MB     │
+└───────┴─────────────┴──────────┴───────────────────┘
+```
+
+### Specific dtypes
+
+As mentioned earlier, while we return `int4` through `float32` by default, any dtype can be used from `float32`, `float16`, `int8`, and `int4`.
+
+To do so, pass them in after specifying `--dtypes`:
+
+```bash
+accelerate estimate-memory bert-base-cased --dtypes float32 float16
+```
+
+```
+┌────────────────────────────────────────────────────┐
+│    Memory Usage for loading `bert-base-cased`      │
+├───────┬─────────────┬──────────┬───────────────────┤
+│ dtype │Largest Layer│Total Size│Training using Adam│
+├───────┼─────────────┼──────────┼───────────────────┤
+│float32│   84.95 MB  │413.18 MB │      1.61 GB      │
+│float16│   42.47 MB  │206.59 MB │     826.36 MB     │
+└───────┴─────────────┴──────────┴───────────────────┘
+```
+
+## Caviats with this calculator
+
+This calculator will tell you how much memory is needed to purely load the model in, *not* to perform inference.
```

Comment:
Maybe it could be emphasized here that "largest layer" and "total size" are exact calculations, whereas "training using Adam" is an estimate?

Reply:
Adjusted